BrowserPicker/App.xaml.cs

```diff
@@ -21,20 +24,69 @@ public App()
 				TargetURL = arguments.Length > 1 ? arguments[1] : null;
 			AppDomain.CurrentDomain.UnhandledException += CurrentDomainOnUnhandledException;
 			ViewModel = new ViewModel(forceChoice);
+		}
+
+		protected override async void OnStartup(StartupEventArgs e)
+		{
+			UnderlyingTargetURL = await GetUnderlyingURLAsync(TargetURL);
 			Deactivated += (sender, args) => ViewModel.OnDeactivated();
+
+			ViewModel.Initialize();
+
+			MainWindow = new MainWindow();
+			MainWindow.Show();
+		}
+
+		private async Task<string> GetUnderlyingURLAsync(string url)
+		{
+			if (url.StartsWith("https://staticsint.teams.cdn.office.net/evergreen-assets/safelinks/"))
+			{
+				var uri = new Uri(url);
+				var queryString = uri.Query;
```

Comment:
Would it not make sense to simply use `HttpUtility.ParseQueryString` here?

Reply:
Oooh, good shout. Updated :-)